### PR TITLE
Create unit tests for HttpResponse classes

### PR DESCRIPTION
**Pull request by llam15**
Add additional checks to HttpRequest unit tests

### CODE REVIEW

File: HttpResponse.cc [hunk 1]
```diff
@@ -89,40 +102,23 @@ std::string FileResponse::get_content_type(const std::string &filename_str) {
         return TYPE_JPEG;
     } else if (type == "gif") {
```
Comment (tyhchoi) [inline HttpResponse.cc:103]: when we check the file type, is it case sensitive?

File: Makefile [hunk 2]
```diff
@@ -25,15 +25,21 @@ config_parser_test: config_parser.cc $(GTEST_CLASSES)
...
 libgtest.a: gtest-all.o
 	ar -rv $@ $^
 
 gtest-all.o: ${GTEST_DIR}/src/gtest-all.cc
 	$(CXX) $(GTEST_FLAGS) -c ${GTEST_DIR}/src/gtest-all.cc
 
 coverage: COVFLAGS += -fprofile-arcs -ftest-coverage
-coverage: Webserver_test config_parser_test HttpRequest_test
-	./Webserver_test && gcov -r Webserver.cc; ./config_parser_test && gcov -r config_parser.cc; ./HttpRequest_test && gcov -r HttpRequest.cc
+coverage: Webserver_test config_parser_test HttpRequest_test HttpResponse_test
```
Comment (tyhchoi) [inline Makefile:38]: Is there a way to test the coverage of one at a time?

File: HttpResponse.cc (hunk 1 above)
Comment (llam15) [reply HttpResponse.cc:103]: I transform it to all lower case before checking the type

File: Makefile (hunk 2 above)
Comment (llam15) [reply Makefile:38]: We could probably make more targets to make it more specific?

File: HttpResponse.cc (hunk 1 above)
Comment (tyhchoi) [reply HttpResponse.cc:103]: ah yea, it just got cut off lol

File: Makefile (hunk 2 above)
Comment (tyhchoi) [reply Makefile:38]: i think we should do it for cc files that we will edit in the future.

**Pull request merged**